### PR TITLE
[Modules] Updated EventHub Namespaces to new dependencies approach & extended readme script 

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated EventHub Namespaces to new dependencies approach
- Extended readme script to work with empty bicep parameter blocks

| Pipeline |
| - |
| [EventHub: Namespaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.eventhub.namespaces.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: utilities/tools/Set-ModuleReadMe.ps1 [hunk 1]
```diff
@@ -649,6 +649,11 @@ function ConvertTo-FormattedJSONParameterObject {
         [string] $BicepParamBlock
     )
 
+    if ([String]::IsNullOrEmpty($BicepParamBlock)) {
```
Comment (eriqua) [inline utilities/tools/Set-ModuleReadMe.ps1:652]: Kindly remove changes not in the scope of this PR

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: Actually it was. It is in the title and description 😉

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: Will merge set change back from main once that PR is merged

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: I will merge back from main once that PR ist merged

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: I will merge back from main once that PR ist merged

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: I wish there'd be a native feature to set one PR as a predecessor to another. As this one would be blocked without the fix that comes with it. Meh... Maybe one day

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1]: done

File: modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      virtualNetworkRules: [
+        {
+          ignoreMissingVnetServiceEndpoint: true
+          subnetResourceId: resourceGroupResources.outputs.subnetResourceId
+        }
+      ]
+    }
+    privateEndpoints: [
+      {
+        privateDnsZoneGroups: {
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep:153]:
```suggestion
        privateDnsZoneGroup: {
```

File: modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    privateEndpoints: [
+      {
+        privateDnsZoneGroups: {
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep:46]:
```suggestion
        privateDnsZoneGroup: {
```

File: modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,175 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.eventhub.namespaces-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/common/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

File: modules/Microsoft.EventHub/namespaces/.test/min/deploy.test.bicep [hunk 7]
```diff
@@ -0,0 +1,36 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.EventHub/namespaces/.test/min/deploy.test.bicep [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.eventhub.namespaces-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/min/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.EventHub/namespaces/.test/min/deploy.test.bicep [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/min/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

File: modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep [hunk 10]
```diff
@@ -0,0 +1,56 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.eventhub.namespaces-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.EventHub/namespaces/.test/pe/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**